# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/html/HTMLImageElement.h
# Commit: acdd013202e1
# Full Hash: acdd013202e1e688986e7e6c747b7c7fee38474e
# Author: Boris Chiou <boris.chiou@gmail.com>
# Date: 2022-08-23 02:46:57
# Regressor Bug: 1694741
# File Overlap Count: 1
# Description:
#   Bug 1694741 - Part 6: Map width/height attributes to the style of img elements. r=emilio
#   
#   This patch will use the width/height attributes from <source> to override
#   width/height/aspect-ratio CSS property values of <img> elements.
#   
# ==============================================================================

diff -r 2b4d9516b882 -r acdd013202e1 dom/html/HTMLImageElement.h
--- a/dom/html/HTMLImageElement.h	Mon Aug 22 20:18:39 2022 +0000
+++ b/dom/html/HTMLImageElement.h	Mon Aug 22 20:18:39 2022 +0000
@@ -270,6 +270,10 @@
 
   void LazyLoadImageReachedViewport();
 
+  // This is used when restyling, for retrieving the extra style from the source
+  // element.
+  const nsMappedAttributes* GetMappedAttributesFromSource() const;
+
  protected:
   virtual ~HTMLImageElement();
 
@@ -310,6 +314,11 @@
   // we don't actually care which changed or to what
   void PictureSourceMediaOrTypeChanged(nsIContent* aSourceNode, bool aNotify);
 
+  // This is called when we update "width" or "height" attribute of source
+  // element.
+  void PictureSourceDimensionChanged(HTMLSourceElement* aSourceNode,
+                                     bool aNotify);
+
   void PictureSourceAdded(HTMLSourceElement* aSourceNode = nullptr);
   // This should be called prior to the unbind, such that nextsibling works
   void PictureSourceRemoved(HTMLSourceElement* aSourceNode = nullptr);
@@ -408,6 +417,8 @@
            GetParentElement()->IsHTMLElement(nsGkAtoms::picture);
   }
 
+  void InvalidateAttributeMapping();
+
   void SetResponsiveSelector(RefPtr<ResponsiveImageSelector>&& aSource);
   void SetDensity(double aDensity);
 